sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlanVisitor.scala

```diff
@@ -47,6 +49,16 @@ trait LogicalPlanVisitor[T] {
 
   def default(p: LogicalPlan): T
 
+  def visitSubqueryExpression(p: LogicalPlan): LogicalPlan = {
+    p.transformExpressionsDown {
```

Comment:
If we don't care about iteration order, `transformExpressions` should be better.

Reply:
> If we don't care about iteration order, `transformExpressions` should be better.

Done